classes/Cart.php

```diff
@@ -2982,11 +3052,55 @@ public function getDeliveryOptionList(Country $default_country = null, $flush =
                     $delivery_option_list[$id_address][$key]['unique_carrier'] = (count($delivery_option_list[$id_address][$key]['carrier_list']) <= 1);
                 }
             }
+
+            $multi_carriers = [];
+
+            foreach ($packages as $id_package => $package) {
+                if (!$multi_carriers) {
+                    foreach ($package['carrier_list'] as $id_carrier) {
+                        $multi_carriers[$id_carrier . ','] = [
+                            'is_best_price' => false,
+                            'is_best_grade' => false,
+                            'unique_carrier' => false,
+                            'carrier_list' => [
+                                $id_carrier => [
+                                    'price_with_tax' => $carriers_price[$id_address][$id_package][$id_carrier]['with_tax'],
+                                    'price_without_tax' => $carriers_price[$id_address][$id_package][$id_carrier]['without_tax'],
+                                    'instance' => $carriers_instance[$id_carrier],
+                                    'package_list' => [$id_package],
+                                    'product_list' => $package['product_list'],
+                                ]
+                            ]
+                        ];
+                    }
+                } else {
+                    foreach ($package['carrier_list'] as $id_carrier) {
+                        foreach ($multi_carriers as $key => $value) {
+                            $new_multi_carriers[$key . $id_carrier . ','] = $value;
+                            $new_multi_carriers[$key . $id_carrier . ',']['carrier_list'][$id_carrier] = [
+                                'price_with_tax' => $carriers_price[$id_address][$id_package][$id_carrier]['with_tax'],
+                                'price_without_tax' => $carriers_price[$id_address][$id_package][$id_carrier]['without_tax'],
+                                'instance' => $carriers_instance[$id_carrier],
+                                'package_list' => [$id_package],
+                                'product_list' => $package['product_list'],
+                            ];
+                        }
+                        $multi_carriers = $new_multi_carriers;
```

Comment:
This code I think it have an error.

$multi_carriers = $new_multi_carriers should go after the second foreach is closed